js/ui/file_manager/ui.file_manager.editing.js

```diff
@@ -160,40 +212,174 @@ class FileManagerEditingControl extends Widget {
             return;
         }
 
-        if(isFunction(action)) {
-            action(arg);
+        if(!action.dialog) {
+            return action.action(arg);
         } else {
-            this._tryEditAction(action, arg);
+            return this._tryEditAction(action, arg);
```

Comment:
This need will make simpler: `!action.dialog ? action.action(arg) : this._tryEditAction(action, arg)`

Reply:
Changed to this version.